### PR TITLE
Fix inability to clear dateFilter fields on source updates

### PR DESCRIPTION
**Pull request by attwad**
It turns out that fields with undefined values are just dropped from the request body by the openapi validator, so we were sending {dateFilter: undefined} and within the source controller updater handler we were seeing req.body == {}

:-(

There doesn't seem to be an option in openapi to keep those fields, there is "nullable" but that just sets the field to null when we save the source document, it doesn't remove it from the document as we want.

So the manual-and-cumbersome alternative that I found was to check and set the field to undefined ourselves if we got an empty {} (non null and non undefined) dateFilter.

Also fixes a typo in the edit case cypress test (how was it working until now??) and better English on the displayed date filter.

For #1198

### CODE REVIEW

File: verification/curator-service/api/src/controllers/sources.ts [hunk 1]
```diff
@@ -97,6 +97,12 @@ export default class SourcesController {
                 });
                 return;
             }
+            // Undefined fields are removed from the request body by openapi
+            // validator, if we want to unset the dateFilter we have to pass an
+            // empty object and set it undefined ourselves here.
```
Comment (allysonjp715) [inline verification/curator-service/api/src/controllers/sources.ts:102]: :/ should we do this for other fields too?

File: verification/curator-service/ui/src/components/SourceTable.tsx [hunk 2]
```diff
@@ -18,7 +18,6 @@ import ParsersAutocomplete from './ParsersAutocomplete';
 import SourceRetrievalButton from './SourceRetrievalButton';
 import TextField from '@material-ui/core/TextField';
 import axios from 'axios';
-import { isUndefined } from 'util';
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/SourceTable.tsx:21]: Is this util used elsewhere or can it be removed?

**Review by allysonjp715 — APPROVED** [review]

File: verification/curator-service/api/src/controllers/sources.ts (hunk 1 above)
Comment (attwad) [reply verification/curator-service/api/src/controllers/sources.ts:102]: This is the only "struct", other singular fields like strings are passed as empty strings and get through as expected.
The automation struct is handled separately.

File: verification/curator-service/ui/src/components/SourceTable.tsx (hunk 2 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/SourceTable.tsx:21]: I can't find any other occurences.
Not sure where this was coming from.

image

**Pull request merged**